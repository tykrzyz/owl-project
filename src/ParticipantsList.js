import Participant from './Participant'
import './ParticipantList.css'

export default function ParticipantList(props){
    let key = 0
    let participantsHTML = props.participants.map(participant => {
        key++
        return <Participant name={participant.name} avatar={participant.avatar} id={participant.id} inSession={participant.inSession} onStage={participant.onStage} key={participant.id}/> 
    })

    return(
        <div className='participants-list'>
            {participantsHTML}
        </div>
    )
}

