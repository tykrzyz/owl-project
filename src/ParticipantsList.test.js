import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ParticipantsList from './ParticipantsList';

const participants = [
    {
        id: 1,
        name: 'Koren Templeton',
        avatar:
            'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: true
    },
    {
        id: 2,
        name: 'Caty Flucker',
        avatar:
            'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: false
    }
]

describe('<ParticipantList />', () => {
    it('should render without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ParticipantsList participants={participants}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(<ParticipantsList participants={participants}/>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
        });
})