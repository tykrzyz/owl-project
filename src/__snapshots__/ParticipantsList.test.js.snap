// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ParticipantList /> renders the UI as expected 1`] = `
<div
  className="participant-list"
>
  <h1>
    <div
      className="participant"
    >
      <div
        className="avatar"
      >
        <img
          alt="Koren Templeton's avatar"
          src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
        />
      </div>
      <div
        className="info"
      >
        <h2>
          Koren Templeton
        </h2>
        <p>
          on stage
        </p>
      </div>
    </div>
    <div
      className="participant"
    >
      <div
        className="avatar"
      >
        <img
          alt="Caty Flucker's avatar"
          src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
        />
      </div>
      <div
        className="info"
      >
        <h2>
          Caty Flucker
        </h2>
        <p>
          in session
        </p>
      </div>
    </div>
  </h1>
</div>
`;
